packages/core/content-releases/server/src/controllers/validation/release-action.ts

```diff
@@ -0,0 +1,43 @@
+import { yup, validateYupSchema, errors } from '@strapi/utils';
+import { RELEASE_MODEL_UID } from '../../constants';
+import { Release, ReleaseActionCreateArgs } from '../../types';
+
+const releaseActionCreateSchema = yup.object().shape({
+  releaseId: yup.number().required(),
+  entry: yup
+    .object()
+    .shape({
+      id: yup.number().required(),
+      contentType: yup.string().required(),
+    })
+    .required(),
+  type: yup.string().oneOf(['publish', 'unpublish']).required(),
+});
+
+export const validateReleaseActionCreateSchema = validateYupSchema(releaseActionCreateSchema);
+
+export async function validateUniqueEntryInRelease(releaseActionArgs: ReleaseActionCreateArgs) {
+  /**
+   * Asserting the type, otherwise TS complains: 'release.actions' is of type 'unknown', even though the types come through for non-populated fields...
+   * Possibly related to the comment on GetValues: https://github.com/strapi/strapi/blob/main/packages/core/types/src/modules/entity-service/result.ts
+   */
+  const release = (await strapi.entityService.findOne(
+    RELEASE_MODEL_UID,
+    releaseActionArgs.releaseId,
+    {
+      populate: { actions: { populate: { entry: { fields: ['id'] } } } },
+    }
+  )) as Release | null;
```

Comment:
❓ Maybe there is a better way to handle this...

Reply:
I am doing the same at the moment 🙃 I think there is no clean alternative at the moment